src/prng/isaac.rs

```diff
@@ -10,53 +10,97 @@
 use core::iter::repeat;
 use core::num::Wrapping as w;
 use core::fmt;
 
 use {Rng, FromRng, SeedableRng};
 
-/// Select 32- or 64-bit variant dependent on pointer size.
-#[cfg(target_pointer_width = "32")]
-pub use prng::IsaacRng as IsaacWordRng;
-#[cfg(target_pointer_width = "64")]
-pub use prng::Isaac64Rng as IsaacWordRng;
-
-#[allow(bad_style)]
-type w64 = w<u64>;
-#[allow(bad_style)]
+#[allow(non_camel_case_types)]
 type w32 = w<u32>;
 
 const RAND_SIZE_LEN: usize = 8;
-const RAND_SIZE: u32 = 1 << RAND_SIZE_LEN;
-const RAND_SIZE_USIZE: usize = 1 << RAND_SIZE_LEN;
+const RAND_SIZE: usize = 1 << RAND_SIZE_LEN;
 
-/// A random number generator that uses the ISAAC algorithm[1].
+/// A random number generator that uses the ISAAC algorithm.
+///
+/// ISAAC stands for "Indirection, Shift, Accumulate, Add, and Count" which are
+/// the principal bitwise operations employed. It is the most advanced of a
+/// series of array based random number generator designed by Robert Jenkins
+/// in 1996[1][2].
+///
+/// Although ISAAC is designed to be cryptographically secure, its design is not
+/// founded in cryptographic theory. Therefore it should _not_ be used for
+/// cryptographic purposes. It is however one of the strongest non-cryptograpic
+/// RNGs, and that while still being reasonably fast.
+///
+/// Where fast random numbers are needed which should still be secure, but where
+/// speed is more important than absolute (cryptographic) security (e.g. to
+/// initialise hashes in the std library), a generator like ISAAC should be
+/// used.
+///
+/// In 2006 an improvement to ISAAC was suggested by Jean-Philippe Aumasson,
+/// named ISAAC+[3]. But because the specification is not complete, there is no
+/// good implementation, and because the suggested bias may not exist, it is not
+/// implemented here.
+///
+/// ## Overview of the ISAAC algorithm:
+/// (in pseudo-code)
+///
+/// ```text
+/// Input: a, b, c, s[256] // state
+/// Output: r[256]         // results
+///
+/// mix(a,i) = a ^ a << 13   i = 0 mod 4
+///            a ^ a >>  6   i = 1 mod 4
+///            a ^ a <<  2   i = 2 mod 4
+///            a ^ a >> 16   i = 3 mod 4
+///
+/// c = c + 1
+/// b = b + c
+///
+/// for i in 0..256 {
+///     x = s_[i]
+///     a = f(a,i) + s[i+128 mod 256]
+///     y = a + b + s[x>>2 mod 256]
+///     s[i] = y
+///     b = x + s[y>>10 mod 256]
+///     r[i] = b
+/// }
+/// ```
+///
+/// Numbers are generated in blocks of 256. This means the function above only
+/// runs once every 256 times you ask for a next random number. In all other
+/// circumstances the last element of the results array is returned.
 ///
-/// The ISAAC algorithm is generally accepted as suitable for
-/// cryptographic purposes, but this implementation has not be
-/// verified as such. Prefer a generator like `OsRng` that defers to
-/// the operating system for cases that need high security.
+/// ISAAC therefore needs comparatively much memory. 2 * 256 * 4 = 2 kb to hold
```

Comment:
ISAAC therefore needs a lot of memory, relative to non-crypto RNGs.

[As I understand, most crypto RNGs need a lot of memory.]